Configuration and Organization Changes

spficklin opened the pull request:
<!-- Please provide the following details when submitting your pull request -->

<!-- What issue number does this PR resolve -->
Issue #198, #158, #135 

## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- Summarize major changes to the code that you made -->
This PR fixes the organizational challenges with the configuration file and the directory structure as described in the issues listed above.   Below is a summary of changes:
- Many of the advanced config settings have been moved out of the primary nextflow.config file into separate files in the `config` folder.
- The input settings in the nextflow.config file have been reorganized into stanzas that are hopefully more intuitive.  
- End-users no longer have to put their files in the `input` folder.  They can put them anywhere they want  and set the full path in the config file rather than having to try and piece together what the paths should be from the other input arguments
- All kallisto, salmon and hisat2 settings are in respective stanzas in the config file, so it should be easier to set everything for a tool in one place.
- The demo data that comes with GEMmaker has been moved to a `demo` folder. The config file comes preset to use the demo data but users can change that. Thus, they no longer have to clean out the `input` folder to run GEMmaker (should prevent common mistakes.
- There is no longer `machine name`, `description` and `project` sections in the input settings. They were too annoying to set.  

## Testing?
<!--- Please describe in detail how to test these changes. -->
Run it once for Hisat2, Kallisto and Salmon. Make sure all of the outpufile are present as with settings in the config file.

JohnHadish commented:
All files are not present. Appears they were not pushed:
```
N E X T F L O W  ~  version 20.10.0
Launching `main.nf` [condescending_pare] - revision: f43375f431
No such file: Config file does not exist: /usr/GEMmaker/config/output.config
```

spficklin commented:
Okay, the output.config file is now present.  Sorry about that.

 I don't recall why we have separate docker setups. It's needed for a reason as it was that way before. I just copied them as they were into appropriate files, but I agree it is confusing.... I'll look and see if  they can be consolidated in some way.

spficklin commented:
Also, I merged in to this branch the recent PR that ben sumbitted removing the e2e pipeline. So there were some merge conflicts I needed to clear out.

spficklin commented:
Okay, I've  merged the docker and signularity options into a single document in the `config/profiles` folder, and I've updated the header comments in the `nextflow.config` file to remove the AnnoTater reference.

I have a few other issues related to documentation so I'll do those in a different pull request.  And as for the Kamiak and Palmetto profiles..., I renamed them kamiak and palmetto because they have hardcoded settings for both.  They never were generic SLURM and PBS profiles.  So they are there as examples not as templates.  I think the best thing to do is update the documentation with instructions for how to configure GEMmaker for execution both on a cluster and on Kubernetes.   If you agree then I think we can merge....

spficklin commented:
Well, on second thought. I think we can make those profiles generic. On second look the only thing that's specific to a cluster is the Kamiak settings, So, I'll take those out.

spficklin commented:
Okay @JohnHadish let me know what you think now. I've renamed the cluster configs back to generic SLURM/PBS and removed the queue names that were specific to our clusters.

JohnHadish commented:
Approved with these additions we discussed:
- `nextflow.config` updated to reflect new generic names (slurm and pbs)
- `slurm.process` queue name should be left blank (will run on back queue) `""`
- `pbs.process` same as slurm (I have not tested this)
